src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -238,19 +238,21 @@ public static synchronized ArrayList getFifoNextSpanData(
   }
 
   /**
-   * Compare a start and end time in milliseconds to see if they are in the same second.
+   * Compare a start and end time in milliseconds to see if enough time has passed for an EBD call.
    *
    * @param startTimeMillis the start time in milliseconds.
    * @param endTimeMillis the end time in milliseconds.
-   * @return true if the times are in the same second
+   * @return true if enough time has passed for the next EBD call
    */
-  private static boolean sameSecond(long startTimeMillis, long endTimeMillis) {
-    boolean isSameSecond = false;
+  private static boolean enoughSeconds(long startTimeMillis, long endTimeMillis) {
+    boolean enoughTimePassed = true;
     long oneSecondMillis = 1000;
-    if ((startTimeMillis / oneSecondMillis) == (endTimeMillis / oneSecondMillis)) {
-      isSameSecond = true;
+    long secondsPassed = (endTimeMillis / oneSecondMillis) - (startTimeMillis / oneSecondMillis);
+
+    if (secondsPassed < 2) {
```

Comment:
2 is a magic number

Reply:
I think there should be a comment associated with the '2' number. It's explained a bit in this PR, but just looking at the code itself does not show why the 2 seconds value was chosen.